jupyterlab_gcedetails/src/components/hardware_scaling_form.tsx

```diff
@@ -202,37 +219,65 @@ export class HardwareScalingForm extends React.Component<Props, State> {
     this.props.onSubmit(configuration);
   }
 
-  private displayPricingEstimation() {
+  private async getOldConfigurationPrice() {
     const { details, priceService } = this.props;
-    const { configuration } = this.state;
 
-    const zone = details.instance.zone.split('/').pop();
-    const currentPrice = priceService.getPrice(zone, this.oldConfiguration);
-    const newPrice = priceService.getPrice(zone, configuration);
-    const priceDifference = (newPrice - currentPrice);
+    const zone = extractLast(details.instance.zone);
+    this.oldConfigurationPrice = await priceService.getPrice(
+      zone,
+      this.oldConfiguration
+    );
+  }
+
+  private async updatePricingEstimation(configuration) {
+    const { details, priceService } = this.props;
+
+    const zone = extractLast(details.instance.zone);
+    const newConfigurationPrice = await priceService.getPrice(
+      zone,
+      configuration
+    );
+    this.setState({
+      newConfigurationPrice,
+    });
+  }
```

Comment:
@prodonjs The reason for the significant refactor in this file is due to the fact that the changes in this commit makes getPrice an asynchronous method. In order to separate fetching the data (by awaiting the asynchronous getPrice method) and displaying it I modified the state of this component to include a newConfigurationPrice variable. This variable's value is updated every time the user modifies a value on the form. I also added a class variable oldConfigurationPrice which stores the price estimation of the original configuration and is only updated once when the component is initialized. The price estimation is only displayed if the values of both these variables are not undefined. Any thoughts on my approach?